.github/workflows/scheduled.yml

```diff
@@ -71,46 +70,161 @@ env:
+
+      - name: Checkout Main
+        if: ${{ steps.get-sig.outputs.stash-hit != 'true' }}
+        uses: actions/checkout@v4
+        with:
+          # hardcode ref without broken pr
+          ref: 'main'
+          path: velox_main
+
+      - name: Build PyVelox
+        if: ${{ steps.get-sig.outputs.stash-hit != 'true' }}
+        env:
+          CCACHE_BASEDIR: "/__w/velox/velox/velox_main"
+        working-directory: velox_main
+        run: |
+          python3 -m venv .venv
+          source .venv/bin/activate
+
+          make python-build
+
+      - name: Create Baseline Signatures
+        if: ${{ steps.get-sig.outputs.stash-hit != 'true' }}
+        working-directory: velox_main
+        run: |
+          source .venv/bin/activate
+          python3 -m pip install deepdiff
+          python3 scripts/signature.py export --spark /tmp/signatures/spark_signatures_main.json
+          python3 scripts/signature.py export --presto /tmp/signatures/presto_signatures_main.json
+
+      - name: Save Function Signature Stash
+        if: ${{ steps.get-sig.outputs.stash-hit != 'true' }}
+        uses: assignUser/stash/save@v1
+        with:
+          path: /tmp/signatures
+          key: function-signatures
 
-      - name: "Checkout Repo"
+      - name: Checkout Contender
         uses: actions/checkout@v4
         with:
           path: velox
           submodules: 'recursive'
           ref: "${{ inputs.ref }}"
 
-      - name: "Install dependencies"
+      - name: Zero Ccache Statistics
         run: |
-          cd  velox
-          source ./scripts/setup-ubuntu.sh
-          ccache -vsz
+          ccache -sz
 
       - name: Build
+        env:
+          CCACHE_BASEDIR: "/__w/velox/velox/velox"
+          EXTRA_CMAKE_FLAGS: "-DVELOX_ENABLE_ARROW=ON -DVELOX_BUILD_PYTHON_PACKAGE=ON ${{ inputs.extraCMakeFlags }}"
         run: |
-          cd velox
-          make debug NUM_THREADS="${{ inputs.numThreads || 8 }}" MAX_HIGH_MEM_JOBS="${{ inputs.maxHighMemJobs || 8 }}" MAX_LINK_JOBS="${{ inputs.maxLinkJobs || 4 }}" EXTRA_CMAKE_FLAGS="-DVELOX_ENABLE_ARROW=ON ${{ inputs.extraCMakeFlags }}"
+          EXTRA_CMAKE_FLAGS="-DPYTHON_EXECUTABLE=$(python3 -c "import sys; print(sys.executable)") $EXTRA_CMAKE_FLAGS"
+          make debug
 
       - name: Ccache after
-        run: ccache -vs
+        run: ccache -s
 
-      - name: "Save ccache"
+      - name: Save ccache
         uses: assignUser/stash/save@v1
         with:
           path: "${{ env.CCACHE_DIR }}"
-          key: ccache-fuzzer
+          key: ccache-fuzzer-centos
+          retention-days: "${{ env.RETENTION }}"
+
+      - name: Build PyVelox
+        env:
+          VELOX_BUILD_DIR: "_build/debug"
+        run: |
+          python3 -m venv .venv
+          source .venv/bin/activate
+          python3 -m pip install -e .
+
+      - name: Create and test new function signatures
+        id: sig-check
+        run: |
+          source .venv/bin/activate
```

Comment:
Maybe we should just move all the logic below to a separate file , what do you say ?

Reply:
I prefer to keep single-use, small'ish scripts like this inline to avoid the indirection of having to look into another file when working on the workflow + additional abstractions like params for the paths used etc.. If this grows anymore or is reused then it would make sense.

Reply:
Actually just remembered that you are working on adding the aggregation sig checks. So yeah makes sense.

Reply:
I added a combo command to signature.py that does all of this. Still needs testing with changed signatures of course.